### PR TITLE
Update to integration tests - support mobile environment

### PR DESCRIPTION
**Pull request by adamjarling**
## What does this do?
- Update tests and fixes a duplicate Header component in layout

## How to test?
- Run the test suite `pnpm test`, or individually in VSCode.  Verify all pass.

### CODE REVIEW

File: playwright.config.ts [hunk 1]
```diff
@@ -51,14 +51,14 @@ export default defineConfig({
...
-    // {
-    //   name: "Mobile Chrome",
-    //   use: { ...devices["Pixel 5"] },
-    // },
-    // {
-    //   name: "Mobile Safari",
-    //   use: { ...devices["iPhone 12"] },
-    // },
+    {
+      name: "Mobile Chrome",
```
Comment (jrgriffiniii) [inline playwright.config.ts:55]: 💯

File: playwright.config.ts [hunk 2]
```diff
@@ -51,14 +51,14 @@ export default defineConfig({
...
+      use: { ...devices["Pixel 5"] },
+    },
+    {
+      name: "Mobile Safari",
```
Comment (jrgriffiniii) [inline playwright.config.ts:59]: 💯

File: tests/layout.spec.ts [hunk 3]
```diff
@@ -8,48 +8,103 @@ const links = [
...
-  await page.getByRole("link", { name: "Contact" }).nth(1).click();
-  await expect(
-    page.getByRole("heading", { name: "Get in touch" }),
-  ).toBeVisible();
-
-  await expect(
-    page.getByRole("link", { name: "Learn more" }).nth(1),
-  ).toHaveAttribute("href", "https://github.com/samvera/hyku");
+  // Test desktop navigation
+  if (!isMobile) {
```
Comment (jrgriffiniii) [inline tests/layout.spec.ts:25]: 👍

File: tests/layout.spec.ts [hunk 4]
```diff
@@ -8,48 +8,103 @@ const links = [
...
+      page.getByRole("heading", { name: "Get in touch" }),
+    ).toBeVisible();
+
+    await expect(
+      page.getByRole("link", { name: "Learn more" }).nth(1),
+    ).toHaveAttribute("href", "https://github.com/samvera/hyku");
+  }
+
+  // Test mobile navigation
+  if (isMobile) {
```
Comment (jrgriffiniii) [inline tests/layout.spec.ts:68]: 👍

**Review by jrgriffiniii — APPROVED** [review]
This is superb, thank you very much for providing this, as well as for establishing this pattern. I can definitely see how this may be easily and readily extended to support additional mobile browsers.

**Comment by adamjarling** [comment]
> This is superb, thank you very much for providing this, as well as for establishing this pattern. I can definitely see how this may be easily and readily extended to support additional mobile browsers.

@jrgriffiniii Great.  Yea, there are a few patterns for targeting mobile environments.  Playwright's browsers come with some popular ones, so it's pretty easy to just plug into what they ship to test against.  

Here are some other potential mobile test patterns I came across, if helpful or you wanted to experiment:

### Pattern 2

```ts
import { test, expect } from '@playwright/test';

test.describe('Mobile tests', () => {
  test.use({ projectName: 'Mobile Chrome' });

  test('Mobile Chrome specific test', async ({ page }, testInfo) => {
    expect(testInfo.project.name).toBe('Mobile Chrome');
    // Mobile Chrome specific tests
  });
});

test.describe('Desktop tests', () => {
  test.use({ projectName: 'chromium' });

  test('Desktop Chrome specific test', async ({ page }, testInfo) => {
    expect(testInfo.project.name).toBe('chromium');
    // Desktop Chrome specific tests
  });
});
```


### Pattern 3 

```ts
import { test, expect } from '@playwright/test';

test('Desktop only test', async ({ page }) => {
  // This test will only run on desktop configurations
});

test('Mobile only test', async ({ page }) => {
  // This test will only run on mobile configurations
});

test.describe('Device specific tests', () => {
  test('iPhone 12 test', async ({ page }, testInfo) => {
    test.skip(testInfo.project.name !== 'Mobile Safari', 'iPhone 12 specific test');
    // iPhone 12 specific test logic
  });

  test('Pixel 5 test', async ({ page }, testInfo) => {
    test.skip(testInfo.project.name !== 'Mobile Chrome', 'Pixel 5 specific test');
    // Pixel 5 specific test logic
  });
});
```


### Fixture Pattern
As you dig more into Playwright, you'll come across their Fixture approach.  It's pretty interesting and powerful.  Here's one generic example:

```ts
import { test as base, expect } from '@playwright/test';

type MyFixtures = {
  isMobile: boolean;
  deviceName: string;
};

const test = base.extend<MyFixtures>({
  isMobile: async ({ }, use, testInfo) => {
    await use(testInfo.project.name.includes('Mobile'));
  },
  deviceName: async ({ }, use, testInfo) => {
    await use(testInfo.project.name);
  },
});

test('Device aware test with custom fixtures', async ({ page, isMobile, deviceName }) => {
  console.log(`Running on: ${deviceName}, Mobile: ${isMobile}`);
  
  if (isMobile) {
    // Mobile specific test logic
  } else {
    // Desktop specific test logic
  }
});
```

**Pull request merged**